Source/WebGPU/WebGPU/Sampler.mm

```diff
@@ -240,7 +241,7 @@ static uint32_t miscHash(MTLSamplerDescriptor* descriptor)
 
     Locker locker { samplerStateLock };
     if (auto it = retainedSamplerStates->find(*m_samplerIdentifier); it != retainedSamplerStates->end()) {
-        it->value.apiSamplerList.remove(*m_samplerIdentifier);
```

Comment:
The issue with the original patch is that UniqueSamplerIdentifier uniquely identifies an object conforming to the id<MTLSampler> protocol and not a Sampler object, so having the HashSet store UniqueSamplerIdentifier is insufficient.

A test was added for this issue originally but 303099@main passed all tests, I need to figure out a way to correctly test it. Likely two workers creating identical Sampler objects would work with API validation layer enabled